Source/EasyNetQ/Logging/LoggerExtensions.cs

```diff
@@ -0,0 +1,523 @@
+using System;
+
+namespace EasyNetQ.Logging
+{
+    /// <summary>
+    ///     Extension methods for the <see cref="ILogger"/> interface.
+    /// </summary>
+    public static class LoggerExtensions
+    {
+        private static readonly object[] EmptyParams = Array.Empty<object>();
+
+        /// <summary>
+        ///     Check if the <see cref="LogLevel.Debug"/> log level is enabled.
+        /// </summary>
+        /// <param name="logger">The <see cref="ILogger"/> to check with.</param>
+        /// <returns>True if the log level is enabled; false otherwise.</returns>
+        public static bool IsDebugEnabled(this ILogger logger)
+        {
+            GuardAgainstNullLogger(logger);
```

Comment:
Consider to switch using NRT instead such null-checks (noisy).

Reply:
Could you elaborate?

Reply:
I've ditched these checks...

Reply:
Ok, we need to adopt this in EasyNetQ codebase separately...

Reply:
Of course. Tons of changes.